internal/resources/app_service_cidr.go

```diff
@@ -0,0 +1,342 @@
+	}
+
+	refreshedState := providerschema.AppServiceCIDR{
+		Id:             types.StringValue(allowListResp.Id),
+		OrganizationId: types.StringValue(organizationId),
+		ProjectId:      types.StringValue(projectId),
+		ClusterId:      types.StringValue(clusterId),
+		AppServiceId:   types.StringValue(appServiceId),
+		Cidr:           types.StringValue(allowListResp.Cidr),
+		Audit:          auditObj,
+	}
+
+	// Set optional fields
+	if allowListResp.Comment != "" {
+		refreshedState.Comment = types.StringValue(allowListResp.Comment)
+	}
+
+	if allowListResp.ExpiresAt != "" {
+		refreshedState.ExpiresAt = types.StringValue(allowListResp.ExpiresAt)
+	}
+
+	return &refreshedState, nil
+}
+
+func (a *AppServiceCidr) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state providerschema.AppServiceCIDR
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Validate parameters were successfully imported
+	organizationId, projectId, clusterId, appServiceId, allowedCIDRId, err := state.Validate()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Validating App Service Allowed CIDR state",
+			"Could not validate App Service Allowed CIDR: "+err.Error(),
+		)
+		return
+	}
+
+	// refresh the existing allow list
+	refreshedState, err := a.refreshAllowedCIDR(ctx, organizationId, projectId, clusterId, appServiceId, allowedCIDRId)
+	if err != errors.ErrNotFound {
+		tflog.Info(ctx, "resource doesn't exist in remote server removing resource from state file")
+		resp.State.RemoveResource(ctx)
+		return
+	} else if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading App Service Allowed CIDR",
+			"Could not read App Service Allowed CIDR "+allowedCIDRId+": "+err.Error(),
+		)
+		return
+	}
+
+	if state.ExpiresAt.ValueString() != refreshedState.ExpiresAt.ValueString() {
+		refreshedState.ExpiresAt = state.ExpiresAt
+	}
+
+	// Set refreshed state
+	diags = resp.State.Set(ctx, refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Update is not supported for App services allowed cidrs.
+func (a *AppServiceCidr) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	// Couchbase Capella's v4 does not support a PUT endpoint for App Services Allowed CIDRs.
+	// App Services Allowed CIDRs can only be created, read and deleted.
+	// https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/Allowed-CIDRs-(App-Services)
+	//
+	// Note: In this situation, terraform apply will default to deleting and executing a new create.
+	// The update implementation should simply be left empty.
+	// https://developer.hashicorp.com/terraform/plugin/framework/resources/update
+}
+
+func (a *AppServiceCidr) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	// Retrieve existing state
```

Comment:
nit

it's obvious from function what is happening.  comment doesn't tell us anything new